src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -400,26 +401,30 @@ public class UTIL_ACCT_Naming {
     * @return String.
     */
     private static String cleanFirstNameFormat(String acctNameFormat) {
+        String firstNameFormat = '';
+
         //Checks {!FirstName}
         String regexPattern = '(?:^|\\W)FirstName(?:$|\\W)';
         Pattern MyPattern = Pattern.compile(regexPattern);
         Matcher MyMatcher = MyPattern.matcher(acctNameFormat);
 
-        //Checks {!{FirstName}}
+        //Checks {!{!FirstName}}
         String regexPatternDoubleToken = '\\{![^\\}]*\\{!.*\\}[^\\{!]*\\}';
         Pattern MyPatternDoubleTokenCheck = Pattern.compile(regexPatternDoubleToken);
         Matcher MyMatcherDoubleTokenCheck = MyPatternDoubleTokenCheck.matcher(acctNameFormat);
 
-        //Set setFNSpec if it finds {!{FirstName}}
         if (MyMatcherDoubleTokenCheck.find()) {
+            //If it finds {!{!FirstName}}
             setFNSpec = true;
-        }
-
-        if (MyMatcher.find()) {
-            return acctNameFormat.subString(MyMatcher.start()-1, MyMatcher.end());
+            firstNameFormat = acctNameFormat.subString(MyMatcherDoubleTokenCheck.start(), MyMatcherDoubleTokenCheck.end());
```

Comment:
Are there unit tests that we can update here to verify the alphabetical order is now fixed?